repo/encryption/pipeline.go

```diff
@@ -0,0 +1,60 @@
+package encryption
+
+import "github.com/kopia/kopia/internal/gather"
+
+// Pipeline creates a pipeline encryptor, where the output of one becomes the input of the next.
+func Pipeline(es ...Encryptor) Encryptor {
```

Comment:
that seems unnecessarily generic, do we need that here and/or that generic? The only use is in `format` package.

Reply:
The idea is to maybe support compression in the future. Also, I though it would be better if forwarding to next one and ecc would be done in different classes. I can move this to the ecc package, and put the code as internal, if you prefer.

Reply:
I'm sceptical we'll be able to do compression like this, since it's conditional and needs to return additional state (whether it was applied or not). 

So we'll have either 1- or 2-stage pipeline which is only used in `format`, so that's why I think we should move it there (on a tactical basis).1-stage pipeline is simply encryptor and we had before was quite reasonable as 2-stage pipeline.